language_detector/main.py

```diff
@@ -4,5 +4,27 @@
 
 def detect_language(text, languages):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    words = text.split()
+    
+    language_guess = ''
+    valid_word_max = 0
+    num_words = {}
+    for language in languages:
+        # The languages are stored in a list of dictionaries
+        # we will go through each language checking for valid words.
+        valid_words = 0
+        for word in words:
+            word = word.lower()
+            word = word.strip(" !@#$%^&*()-_+={}[]|\:;'<>?,./\"")
```

Comment:
That might be true for the sample text in the tests. But in real life there is a chance that you will get common words with punctuation signs which will not make the match. For example in the text `"I knew it! Exclaimed Bob, I always knew it."`. The word `it` would not be found in common using a simple `.split()` without stripping away the extra characters.